### PR TITLE
Add new chaco.plots submodule

### LINKED ISSUE #700: Consolidate plots into a new "chaco.plots"?
- [`BarPlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/barplot.py#L41)
- [`CandlePlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/candle_plot.py#L26)
- [`CMapImagePlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/cmap_image_plot.py#L20)
- [`ColorBar`](https://github.com/enthought/chaco/blob/1ed123cc65d4fd2b9a53a238cd584f8c5dfcac68/chaco/color_bar.py#L28)
- [`ColormappedSegmentPlot`](https://github.com/enthought/chaco/blob/1ed123cc65d4fd2b9a53a238cd584f8c5dfcac68/chaco/segment_plot.py#L434)
- [`ColorMappedScatterPlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/colormapped_scatterplot.py#L46)
- [`QuiverPlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/quiverplot.py#L12)
- [`ContourLinePlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/contour_line_plot.py#L17)
- [`ContourPolyPlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/contour_poly_plot.py#L16)
- [`ErrorBarPlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/errorbar_plot.py#L16)
- [`FilledLinePlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/filled_line_plot.py#L16)
- [`HorizonPlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/horizon_plot.py#L28)
- [`ImagePlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/image_plot.py#L54)
- [`JitterPlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/jitterplot.py#L11)
- [`LinePlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/lineplot.py#L35)
- [`LineScatterPlot1D`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/line_scatterplot_1d.py#L17)
- [`MultiLinePlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/multi_line_plot.py#L34)
- [`PolarLineRenderer`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/polar_line_renderer.py#L16)
- [`PolygonPlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/polygon_plot.py#L17)
- [`ScatterPlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/scatterplot.py#L198)
- [`ScatterPlot1D`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/scatterplot_1d.py#L17)
- [`SegmentPlot`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/segment_plot.py#L24)
- [`TextPlot`](https://github.com/enthought/chaco/blob/1ed123cc65d4fd2b9a53a238cd584f8c5dfcac68/chaco/text_plot.py#L20)
- [`TextPlot1D`](https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/chaco/text_plot_1d.py#L20)

and the respective baseclasses. Ref https://github.com/enthought/chaco/blob/4b3747d407527dde50af6214cad1e6cc2eb6f7b0/docs/source/user_manual/modules_and_classes.rst#renderers

### PR DESCRIPTION
**Pull request by aaronayres35**
This PR would close #700 

It is meant to demonstrate the plan of action for part of the proposed re-org.
If we decide to go ahead with this re-org this draft PR can be marked ready for review / fixed up and merged.

### CODE REVIEW

File: chaco/tests/test_jitterplot.py [hunk 1]
```diff
@@ -8,10 +8,10 @@
 from chaco.api import (
     ArrayDataSource,
     DataRange1D,
+    JitterPlot,
     LinearMapper,
     PlotGraphicsContext,
 )
-from chaco.jitterplot import JitterPlot
```
Comment (aaronayres35) [inline chaco/tests/test_jitterplot.py:14]: I realized this is a common practice in chaco:
in a test module for ____ importing everything except ___ from `chaco.api` and then ___ from the source.  I do not know why this is (hence me undoing that here and just importing `JitterPlot` from the api module).

If it is purposeful I am happy to undo this change.  If it is not (i.e maybe test was written before the class was available in the api module?) I will leave it, and make the similar changes elsewhere in the code base.

File: chaco/candle_plot.py [hunk 2]
```diff
@@ -1,191 +1,19 @@
...
+import warnings
+
+from chaco.plots.candle_plot import CandlePlot  # noqa: F401
+
+warnings.warn(
+    "This module has been moved to sit in chaco/plots and this stub module has"
+    " been kept for backwards compatibility. Importing from this module is"
+    " deprecated, please import needed objects from chaco.api instead",
+    DeprecationWarning
+)
```
Comment (aaronayres35) [inline chaco/candle_plot.py:19]: Note: "Import-time deprecation warnings risk being hidden from test runs."
We may want to update how we are raising these deprecation warnings...

Comment (aaronayres35) [reply chaco/candle_plot.py:19]: We should also update the warning message to be explicit that these modules will be removed in chaco V6 or V7 once that is decided on

File: chaco/api.py [hunk 3]
```diff
@@ -291,7 +291,7 @@
 from .color_mapper import ColorMapper, ColorMapTemplate
 from .discrete_color_mapper import DiscreteColorMapper
 from .transform_color_mapper import TransformColorMapper
-from .horizon_plot import BandedMapper
+from .plots.horizon_plot import BandedMapper
```
[the remark this answers is not in the document]
Comment (aaronayres35) [reply chaco/api.py:294]: Done in the latest commit.  The only things I am unsure about are `ColorBar`, `render_markers`, and also `BandedMapper` from `chaco.plots.horizon_plot`.  In other words which section should they go in for `chaco.api`, should `chaco.plots.api` be divided into sections to separate them out?  Should they be exposed in the `chaco.plots.api` at all, or should `chaco.plots.api` be kept just for `-___Plot___` classes?

**Comment by aaronayres35** [comment]
I am going to convert this PR from a draft PR to a real PR as we do aim to merge it eventually now

File: chaco/plot.py [hunk 4]
```diff
@@ -17,34 +17,34 @@
 from .array_data_source import ArrayDataSource
 from .array_plot_data import ArrayPlotData
 from .base_xy_plot import BaseXYPlot
-from .barplot import BarPlot
-from .candle_plot import CandlePlot
-from .colormapped_scatterplot import ColormappedScatterPlot
-from .contour_line_plot import ContourLinePlot
-from .contour_poly_plot import ContourPolyPlot
-from .cmap_image_plot import CMapImagePlot
+from .plots.barplot import BarPlot
```
Comment (rahulporuri) [inline chaco/plot.py:20]: nitpick - can we reorder the imports here? It looks like these imports were originally ordered - and then a few new plot types were added which broke the order - and now we have broken it once more by moving the modules around

File: chaco/scatterplot.py [hunk 5]
```diff
@@ -1,607 +1,20 @@
...
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+import warnings
+
+from chaco.plots.scatterplot import ScatterPlot  # noqa: F401
```
Comment (rahulporuri) [inline chaco/scatterplot.py:12]:
```suggestion
from chaco.plots.scatterplot import render_markers, ScatterPlot, ScatterPlotView  # noqa: F401
```

File: chaco/segment_plot.py [hunk 6]
```diff
@@ -1,436 +1,20 @@
...
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+import warnings
+
+from chaco.plots.segment_plot import SegmentPlot  # noqa: F401
```
Comment (rahulporuri) [inline chaco/segment_plot.py:12]:
```suggestion
from chaco.plots.segment_plot import ColormappedSegmentPlot, SegmentPlot  # noqa: F401
```

File: chaco/plots/api.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+- :class:`~.TextPlot`
+- :class:`~.TextPlot1D`
+
+"""
+
+from chaco.plots.barplot import BarPlot
+from chaco.plots.candle_plot import CandlePlot
+from chaco.plots.cmap_image_plot import CMapImagePlot
+from chaco.plots.color_bar import ColorBar
+from chaco.plots.colormapped_scatterplot import ColormappedScatterPlot
```
Comment (rahulporuri) [inline chaco/plots/api.py:43]:
```suggestion
from chaco.plots.colormapped_scatterplot import ColormappedScatterPlot, ColormappedScatterPlotView
```

File: chaco/plots/api.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+from chaco.plots.contour.contour_line_plot import ContourLinePlot
+from chaco.plots.contour.contour_poly_plot import ContourPolyPlot
+from chaco.plots.errorbar_plot import ErrorBarPlot
+from chaco.plots.filled_line_plot import FilledLinePlot
+from chaco.plots.horizon_plot import HorizonPlot
```
Comment (rahulporuri) [inline chaco/plots/api.py:48]:
```suggestion
from chaco.plots.horizon_plot import BandedMapper, HorizonPlot
```

File: chaco/plots/api.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+from chaco.plots.image_plot import ImagePlot
+from chaco.plots.jitterplot import JitterPlot
+from chaco.plots.line_scatterplot_1d import LineScatterPlot1D
+from chaco.plots.lineplot import LinePlot
+from chaco.plots.multi_line_plot import MultiLinePlot
+from chaco.plots.polar_line_renderer import PolarLineRenderer
+from chaco.plots.polygon_plot import PolygonPlot
+from chaco.plots.quiverplot import QuiverPlot
+from chaco.plots.scatterplot import render_markers, ScatterPlot
```
Comment (rahulporuri) [inline chaco/plots/api.py:57]:
```suggestion
from chaco.plots.scatterplot import render_markers, ScatterPlot, ScatterPlotView
```

File: chaco/plots/api.py [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+from chaco.plots.scatterplot_1d import ScatterPlot1D
+from chaco.plots.segment_plot import SegmentPlot
```
Comment (rahulporuri) [inline chaco/plots/api.py:59]:
```suggestion
from chaco.plots.segment_plot import ColormappedSegmentPlot, SegmentPlot
```

File: chaco/api.py [hunk 11]
```diff
@@ -325,39 +325,45 @@
...
-from .text_plot_1d import TextPlot1D
-from .segment_plot import SegmentPlot
-from .text_plot import TextPlot
-from .polar_line_renderer import PolarLineRenderer
+
+from chaco.plots.api import (
+    BarPlot,
+    CandlePlot,
+    CMapImagePlot,
+    ColormappedScatterPlot,
```
Comment (rahulporuri) [inline chaco/api.py:342]:
```suggestion
    ColormappedScatterPlot,
    ColormappedScatterPlotView,
    ColormappedSegmentPlot,
```

File: chaco/api.py [hunk 12]
```diff
@@ -325,39 +325,45 @@
...
+    ImagePlot,
+    JitterPlot,
+    LineScatterPlot1D,
+    LinePlot,
+    MultiLinePlot,
+    PolarLineRenderer,
+    PolygonPlot,
+    QuiverPlot,
+    render_markers,
+    ScatterPlot,
```
Comment (rahulporuri) [inline chaco/api.py:357]:
```suggestion
    ScatterPlot,
    ScatterPlotView,
```

File: chaco/plots/api.py (hunk 7 above)
Comment (rahulporuri) [reply chaco/plots/api.py:43]: can you also include this - and other missing imports i suggested in this module - to the module docstring?

File: chaco/api.py (hunk 11 above)
Comment (rahulporuri) [reply chaco/api.py:342]: can you also include this - and other missing imports i suggested in this module - to the module docstring?

File: ci/edmtool.py [hunk 13]
```diff
@@ -121,6 +121,29 @@
 
 doc_ignore = {
     "*/tests",
+    "chaco/barplot.py",
```
Comment (rahulporuri) [inline ci/edmtool.py:124]: It'd be good to add a reason here as to why these are ignored. I think we can create an issue - regarding the removal of the stubs - and link that here.

File: chaco/plot.py [hunk 14]
```diff
@@ -17,34 +17,34 @@
...
+from .plots.candle_plot import CandlePlot
+from .plots.colormapped_scatterplot import ColormappedScatterPlot
+from .plots.contour.contour_line_plot import ContourLinePlot
+from .plots.contour.contour_poly_plot import ContourPolyPlot
+from .plots.cmap_image_plot import CMapImagePlot
+from .plots.filled_line_plot import FilledLinePlot
+from .plots.image_plot import ImagePlot
+from .plots.jitterplot import JitterPlot
 from .plot_label import PlotLabel
```
Comment (rahulporuri) [inline chaco/plot.py:38]: nit-pick : this `plot_label` import is out of order 😅 
`plot_label` would come before `plots.*` imports I think

**Review by rahulporuri — APPROVED** [review]
LGTM with a couple of minor comments.

**Pull request merged**